app/2.0/docs/devguide/dom-template.md

```diff
@@ -60,106 +59,183 @@ except for testing. For caveats about defining elements in the main document,
-    static get template() {
-      if (!memoizedTemplate) {
-        // create a clone of superclass template (`true` = "deep" clone)
-        memoizedTemplate = MySuperClass.template.cloneNode(true);
-        // add a node to the template.
-        let div = document.createElement('div');
-        div.textContent = 'Hello from an extended template.'
-        memoizedTemplate.content.appendChild(div);
-      }
-      return memoizedTemplate;
-    }
+class BaseClass extends Polymer.Element {
+  static get template() {
+    return html`This content has been inherited from BaseClass.`;
   }
+}
+```
 
-})();
+Child class definition { .caption }
+```js
+class ChildClass extends Polymer.Element {
+
+}
 ```
 
-The following example shows how an element could wrap a superclass template with its own
-template.
+[See a working example in Plunker](http://plnkr.co/edit/vS99al?p=preview).
 
-```html
-<dom-module id="my-ext">
-    <template>
-      <h2>Extended template</h2>
-      <!-- superclass template will go here -->
-      <div id="footer">
-        Extended template footer.
-      </div>
-    </template>
-  <script>
-    (function() {
-      let memoizedTemplate;
+#### Override a template definition {#overridebase}
+
+Base class definition { .caption }
 
-      class MyExtension extends MySuperClass {
+```js
+const html = Polymer.html;
 
-        static get is() { return 'my-ext'}
-        static get template() {
-          if (!memoizedTemplate) {
-            // Retrieve this element's dom-module template
-            memoizedTemplate = Polymer.DomModule.import(this.is, 'template');
+class BaseClass extends Polymer.Element {
+  static get template() {
+    return html`This content has been inherited from BaseClass.`;
+  }
+}
+```
 
-            // Clone the contents of the superclass template
-            let superTemplateContents = document.importNode(MySuperClass.template.content, true);
+Child class definition { .caption }
+```js
+const html = Polymer.html;
 
-            // Insert the superclass contents
-            let footer = memoizedTemplate.content.querySelector('#footer');
-            memoizedTemplate.content.insertBefore(superTemplateContents, footer);
-          }
-          return memoizedTemplate;
-        }
-      }
-      customElements.define(MyExtension.is, MyExtension);
-    })();
-  </script>
-</dom-module>
+class ChildClass extends BaseClass {
+  static get template() {
+    return html`
+      Base class template has been overridden. Hello from ChildClass!
+      `;
+  }
+}
+```
+
+[See a working example in Plunker](http://plnkr.co/edit/VCwhuA?p=preview).
+
+#### Extend a base class template {#extendbase}
+
+Base class definition { .caption }
+
+```js
+const html = Polymer.html;
+
+class BaseClass extends Polymer.Element {
+  static get template() {
+    return html`
+      This content has been inherited from BaseClass.
+      `;
+  }
+}
+```
+
+Child class definition { .caption }
+```js
```

Comment:
Arguably this is simpler than using the existing template manipulation code, but it involves some magic that should probably be explained. 

1) The `${expression}` inside a template literal interpolates the value of an expression. (This is basic ES6, and maybe it doesn't need to be spelled out, but if so, we should at least link to the MDN description of template literals at the first reference (in the string template section), because we're using a whole bunch of their features here and not everyone is familiar with them yet.)
2)  Because the getter is static, `super.template` accesses the template property on the superclass constructor.
3) The `Polymer.html` function checks each expression value. If the value is a string, the string is interpolated directly. If the expression is an `HTMLTemplateElement`, the `innerHTML` of the template is interpolated. This makes it very simple to combine existing templates.

Reply:
And this is a lot of words, and maybe should be pushed up to some concepts section somewhere, but I think it's important because we're using a lot of ES6-fu here that seems completely transparent to the @justinfagnani s of the world...